London_11 | Saqib_Javed | Module-User-Focused-Data | Form-Control | week2

saqibjvd opened the pull request:
<!--

You must title your PR like this:

COHORT_NAME | FIRST_NAME LAST_NAME | REPO_NAME | WEEK

For example,

NW4 | Carol Owen | HTML-CSS-Module | Week1

Complete the task list below this message.
If your PR is rejected, check the task list.

-->

## Learners, PR Template

Self checklist

- [ x] I have committed my files one by one, on purpose, and for a reason
- [x ] I have titled my PR with COHORT_NAME | FIRST_NAME LAST_NAME | REPO_NAME | WEEK 
- [x ] I have tested my changes
- [x ] My changes follow the [style guide](https://syllabus.codeyourfuture.io/guides/code-style-guide/)
- [x ] My changes meet the [requirements](./README.md) of this task

## Changelist

Briefly explain your PR.

## Questions

Ask any questions you have for your reviewer.

saqibjvd committed:
basic form layout

WIP. made simple from, tested for validation and any errors. added name to footer section

saqibjvd committed:
CSS

added some css syle to form

saqibjvd committed:
Form control

form control completed, light house score 100. also fixed some files.

saqibjvd committed:
updated html

remove unwanted tag

jamesbaskerville commented on Form-Controls/index.html:
```diff
@@ -9,19 +9,67 @@
-  </form>
+  <div class="form">
+    <label for="fname">First Name:</label><br>
+    <input type="text" id="fname" name="fname" pattern="[A-Za-z]+" title="Only letters are allowed" required/>
```
Nice use of `pattern`! One thing to consider -- names vary widely and may not include characters beyond ASCII or not necessarily be easily splittable into first/last names.

See these blog posts for more assumptions that aren't true about names:
- [What's in a name](https://dev.to/carlymho/whats-in-a-name-validation-4b41)
- [Falsehoods Programmers Believe About Names](https://www.kalzumeus.com/2010/06/17/falsehoods-programmers-believe-about-names/)

Any thoughts on this, e.g. changing the pattern and/or combining the name fields?

saqibjvd replied:
should i do full name insread of first name and last name?

saqibjvd replied:
fixed

jamesbaskerville commented on Form-Controls/index.html:
```diff
@@ -9,19 +9,67 @@
+    <input id="email" type="email" required/>
+  </div>
+
+  <div class="shirt-colours">
```
I notice that I can submit the form without choosing a colour option. How should we handle this case?

saqibjvd replied:
fixed

jamesbaskerville commented on Form-Controls/index.html:
```diff
@@ -9,19 +9,67 @@
+    </fieldset>
+  </div>
+
+  <div class="shirt-size">
```
I notice that I can submit the form without choosing a size option. How should we handle this case?

saqibjvd replied:
fixed

jamesbaskerville commented on Form-Controls/styles.css:
```diff
@@ -0,0 +1,66 @@
+* {
+    padding: 5px;
+    margin: 10px;
+    background-color: rgb(222, 206, 206);
```
assigning a background color to all elements might get tricky -- what's a way we can style the layout's background without doing so for every element?

jamesbaskerville commented on Form-Controls/styles.css:
```diff
@@ -0,0 +1,66 @@
+    justify-content: center;
+  }
+  
+#size{
+    background-color: aliceblue;
+}
```
typically we would avoid styling based on IDs in favor of classes. Why do we need to style the select background like this?

saqibjvd replied:
i just change to class from ID. i thought  i should use id because its unique.

jamesbaskerville reviewed (changes requested):
Really nice job! I like the look of the page a lot -- just a few comments on expected validations and styling best practices.

saqibjvd commented on Form-Controls/index.html:
```diff
@@ -9,19 +9,67 @@
   <form>
-    <!-- write your html here-->
-    <!-- try writing out the requirements first-->
-  </form>
```
fixed

saqibjvd committed:
form control

I updated the HTML and CSS. I made changes before, but I'm not sure why they didn't show up in the PR

saqibjvd committed:
Merge branch 'main' into Form-Controls
